packages/shared/src/retry.ts

```diff
@@ -0,0 +1,108 @@
+type Milliseconds = number;
+
+type RetryOptions = Partial<{
+  /**
+   * The initial delay before the first retry.
+   * @default 125
+   */
+  firstDelay: Milliseconds;
+  /**
+   * The maximum delay between retries.
+   * If set to 0, the delay will increase indefinitely.
+   * @default 0
+   */
+  maxDelay: Milliseconds;
```

Comment:
Only knowing the name (and not reading the jsdoc comment) it's not really clear if this means the max. allowed delay between retries or if this is the max. allowed duration of all retries. Also what happens when this maximum value is passed?

Reply:
Updated the comment to indicate that `The delay between retries will never exceed this value.` and updated the prop name to `maxDelayBetweenRetries`